torch/jit/annotations.py

```diff
@@ -320,7 +320,7 @@ def try_ann_to_type(ann, loc):
             torch.jit._script._recursive_compile_class(ann, loc)
         return EnumType(_qualified_name(ann), get_enum_value_type(ann, loc), list(ann))
     if inspect.isclass(ann):
-        if hasattr(ann, "__torch_script_class__"):
+        if hasattr(ann, "__torch_script_class__") and len(ann.mro()) <= 2:
```

Comment:
Won't this break `torch.no_grad` ? I think it would be more intuitive to switch to storing map of {python class pointer, jit class} instead of annotating the class

Reply:
We have a map from `qualified_name` -> `cls` (Python class, not JIT type) in `_state.py` for all classes that have truly been scripted; what if we changed this to
```suggestion
        if torch.jit._state._get_script_class(ann):
```
In the example that motivated this fix, `A` would be in `_state._script_classes`, but `B` would not.

Reply:
yea sgtm

Reply:
Although I realize now that we're in a weird place where inheritance is prohibited if the class is scripted directly, but allowed if it is scripted implicitly (e.g. through `try_ann_to_type`) and doesn't use any features of inheritance (like `super()`).